tutorials/neurips_challenge_quickstart.md

```diff
@@ -0,0 +1,201 @@
+
+&nbsp;
+
+## Finetuning
+
+[Low-rank Adaptation (LoRA)](https://lightning.ai/pages/community/tutorial/lora-llm/) is a good choice for a first finetuning run. The Dolly dataset has ~15k samples, and the finetuning might take half an hour. 
+
+To accelerate this for testing purposes, edit the [./finetune/lora.py](https://github.com/Lightning-AI/lit-gpt/blob/main/finetune/lora.py) script and change `max_iters = 50000` to `max_iters = 500` at the top of the file.
+
+The following command finetunes the model:
+
+```bash
+CUDA_VISIBLE_DEVICES=2 python finetune/lora.py \
+  --data_dir data/dolly-stablelm3b \
+  --checkpoint_dir checkpoints/stabilityai/stablelm-base-alpha-3b \
+  --out_dir out/stablelm3b/dolly/lora/experiment1 \
+  --precision "bf16-true"
+```
+
+With 500 iterations, this takes approximately 1-2 min on an A100 and uses 26.30 GB GPU memory.
+
+If you are using an RTX 4090, change `micro_batch_size=4` to `micro_batch_size=1` so that the model will only use 12.01 GB of memory.
+
+(More finetuning settings are explained [here](https://lightning.ai/pages/community/tutorial/neurips2023-llm-efficiency-guide/#toc10).)
+
+&nbsp;
+
+## Local Evaluation
+
+The official Lit-GPT competition will use HELM subtasks for model evaluation. 
+
+HELM is currently also being integrated into Lit-GPT to evaluate LLMs before submission. 
+
+However, a tool with a more convenient interface is Eleuther AI's Evaluation Harness, which contains some tasks, for example, TruthfulQA and Gsm8k, that overlap with HELM. We can set up the Evaluation Harness as follows:
+
+```
+cd ..
+git clone https://github.com/EleutherAI/lm-evaluation-harness
+cd lm-evaluation-harness
+pip install -e .
+cd ../lit-gpt
+```
+
+And then we can use it via the following command: 
+
+```bash
+python eval/lm_eval_harness.py \
+  --checkpoint_dir checkpoints/stabilityai/stablelm-base-alpha-3b \
+  --precision "bf16-true" \
+  --eval_tasks "[truthfulqa_mc,gsm8k]" \
+  --batch_size 4 \
+  --save_filepath "results-stablelm-3b.json"
+```
+
+(You can find a full task list in the task table [here](https://github.com/EleutherAI/lm-evaluation-harness/blob/master/docs/task_table.md).)
+
+To evaluate a LoRA-finetuned model, use `lm_eval_harness_lora.py` instead of `eval/lm_eval_harness.py`.
+
+
+&nbsp;
+
+## Submission
+
+You will be required to submit a Docker image for the submission itself. Fortunately, the organizers have a GitHub repository with the exact steps [here](https://github.com/llm-efficiency-challenge/neurips_llm_efficiency_challenge) and a toy-submission setup guide to test your model locally before submission.
+
+
+&nbsp;
+
+## Additional Information & Resources
+
+- [The official NeurIPS 2023 LLM Efficiency Challenge competition website](https://llm-efficiency-challenge.github.io/)
+- A more extensive guide, including environment setup tips: [The NeurIPS 2023 LLM Efficiency Challenge Starter Guide](The NeurIPS 2023 LLM Efficiency Challenge Starter Guide)
```

Comment:
```suggestion
- A more extensive guide, including environment setup tips: [The NeurIPS 2023 LLM Efficiency Challenge Starter Guide](https://lightning.ai/pages/community/tutorial/neurips2023-llm-efficiency-guide)
```

Reply:
good catch!

Reply:
Thanks for the review! I love the attention to detail and really appreciate it!